src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -2413,6 +2476,12 @@ internal bool LoadModuleManifestData(ExternalScriptInfo scriptInfo, ManifestProc
             manifestInfo.PowerShellVersion = powerShellVersion;
             manifestInfo.ProcessorArchitecture = requiredProcessorArchitecture;
             manifestInfo.Prefix = resolvedCommandPrefix;
+
+            if (expFeatureList != null)
+            {
+                manifestInfo.ExperimentalFeatures = new ReadOnlyCollection<ExperimentalFeature>(expFeatureList);
```

Comment:
Just to check, the `PSModuleInfo` will have `ExperimentalFeatures` set to `null` if the field is not present, and is allowed to be an empty collection, but we don't expect anyone to use that.

Asking because it's different to `CompatiblePSEditions` (for example), where not giving the field means we put an empty collection on the `PSModuleInfo` rather than `null`.

I prefer using `null` because it reflects the actual manifest, but want to note that it's still maybe inconsistent with other choices for the `PSModuleInfo` object.

Reply:
Please take a look at `PSModuleInfo.cs`, where the property is defined. The property is given a default value -- a shared static empty read-only collection. So it's always a read-only collection, even if it's not set in the manifest.

Reply:
Ah ok. The way we set that variable is different to CompatiblePSEditions - there the PSModuleInfo is initialised with the empty collection and the property/field itself is read-only (and is also a read-only collection). The way it's set is to add members to it, as opposed to here where a new collection object is set to the field.

Reply:
More strictly, `CompatiblePSEditions` should also return a read-only collection instead of a `List<string>` as an `IEnumerable<string>`. With the current implementation, users can always change this property.